### PR TITLE
make: make `clean` more robust

### PR DESCRIPTION
**Pull request by mor1**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,6 +46,6 @@ build:
 	cd src && make build
 
 clean:
-	cd src && make clean
+	[ -r src/Makefile ] && ( cd src && make clean ) || true
```
Comment (hannesm) [inline Makefile]: can't you just `-cd src && make clean` (see https://www.gnu.org/software/make/manual/make.html#Errors)

Comment (mor1) [inline Makefile]: Oh, probably-- didn't know that. Is it a GNUism or a standard Make thing?
(Do we care? :)

Comment (yallop) [inline Makefile]: or even

```shell
	-make -C src clean
```

Comment (mor1) [inline Makefile]: ack; updated again...

**Pull request merged**